src/requirementslib/models/requirements.py

```diff
@@ -850,23 +908,99 @@ def as_pipfile(self):
-                self._ireq = InstallRequirement.from_line(ireq_line)
-        return self._ireq
+        return self.as_ireq()
+
+    def get_dependencies(self, sources=None):
+        """Retrieve the dependencies of the current requirement.
+
+        Retrieves dependencies of the current requirement.  This only works on pinned
+        requirements.
+
+        :param sources: Pipfile-formatted sources, defaults to None
+        :param sources: list[dict], optional
+        :return: A set of requirement strings of the dependencies of this requirement.
+        :rtype: set(str)
+        """
+        if not sources:
+            sources = [{
+                'name': 'pypi',
+                'url': 'https://pypi.org/simple',
+                'verify_ssl': True,
+            }]
+        return get_dependencies(self.as_ireq(), sources=sources)
+
+    def get_abstract_dependencies(self, sources=None):
+        """Retrieve the abstract dependencies of this requirement.
+
+        Returns the abstract dependencies of the current requirement in order to resolve.
+
+        :param sources: A list of sources (pipfile format), defaults to None
+        :param sources: list, optional
+        :return: A list of abstract (unpinned) dependencies
+        :rtype: list[ :class:`~requirementslib.models.dependency.AbstractDependency` ]
+        """
+
+        if not self.abstract_dep:
+            parent = getattr(self, 'parent', None)
+            self.abstract_dep = AbstractDependency.from_requirement(self, parent=parent)
+        if not sources:
+            sources = [{'url': 'https://pypi.org/simple', 'name': 'pypi', 'verify_ssl': True},]
+        if is_pinned_requirement(self.ireq):
+            deps = self.get_dependencies()
+        else:
+            ireq = sorted(self.find_all_matches(), key=lambda k: k.version)
+            deps = get_dependencies(ireq.pop(), sources=sources)
+        return get_abstract_dependencies(deps, sources=sources, parent=self.abstract_dep)
+
+    def find_all_matches(self, sources=None, finder=None):
+        """Find all matching candidates for the current requirement.
+
+        Consults a finder to find all matching candidates.
+
+        :param sources: Pipfile-formatted sources, defaults to None
+        :param sources: list[dict], optional
+        :return: A list of Installation Candidates
+        :rtype: list[ :class:`~pip._internal.index.InstallationCandidate` ]
+        """
+        if not finder:
+            finder = get_finder(sources=sources)
+        return find_all_matches(finder, self.as_ireq())
+
+    def merge_markers(self, markers):
+        if not isinstance(markers, Marker):
+            markers = Marker(markers)
+        _markers = set(Marker(self.ireq.markers)) if self.ireq.markers else set(markers)
+        _markers.add(markers)
+        new_markers = Marker(" or ".join([str(m) for m in sorted(_markers)]))
+        self.markers = str(new_markers)
+        self.req.req.markers = new_markers
```

Comment:
Note that currently this is not a `packaging.requirements.Requirement` but rather a `requirement-parser.requirements.Requirement` (completely different, but I think we can axe this now)